pinc/Project.inc

```diff
@@ -955,83 +979,38 @@ class Project
 
     // -------------------------------------------------------------------------
 
-    // These should be treated as constants.
-    // (PHP doesn't appear to allow constants as class members.)
-    public $CBP_OKAY = 0;
-    public $CBP_PROJECT_NOT_IN_ROUND = 1;
-    public $CBP_PROJECT_NOT_AVAILABLE = 2;
-    public $CBP_REQUESTOR_NOT_LOGGED_IN = 3;
-    public $CBP_USER_NOT_QUALIFIED_FOR_ROUND = 4;
-    public $CBP_PROJECT_REQUIRES_MAINTENANCE = 5;
-
     public function can_be_proofed_by_current_user()
```

Comment:
A `can_*` or `is_*` function should return an answer to the question it's asked, not an exception as an answer. We can rename this function to `validate_user_can_get_pages_in_project()` which should either pass or return an exception though.

Or the `validate_()` function can call this one and raise the appropriate exception.